lib/rspec/core/reporter.rb

```diff
@@ -79,6 +88,17 @@ def message(message)
       notify :message, Notifications::MessageNotification.new(message)
     end
 
+    # @param event [Symbol] Name of the custom event to trigger on formatters
+    # @param options [Hash] Hash of arguments to provide via `CustomNotification`
+    #
+    # Publish a custom event to supporting registered formatters.
+    # @see RSpec::Core::Notifications::CustomNotification
+    def publish(event, options={})
+      event_name = event.to_sym
```

Comment:
To ensure sending a string param matching an RSpec event doesn't accidentally go through.